x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -953,6 +953,11 @@ export class AlertsClient {
         ),
         updatedBy: username,
         updatedAt: new Date().toISOString(),
+        executionStatus: {
+          status: 'pending',
+          lastExecutionDate: new Date().toISOString(),
```

Comment:
nit: Does it make sense to reset the `lastExecutionDate` here? I guess it would quickly be overwritten the next time the rule runs, but technically, this date is not the last time the rule was executed.

Reply:
It looks like we do set that field to the same "current date" for other cases of `status: 'pending'`.  So, it's consistent.  I think we picked the name `lastExecutionDate` before starting to use `pending`, and so ... the name kinda doesn't make sense any more.  Would be better described (I think) as "lastStateUpdateDate" or something.

Reply:
>Would be better described (I think) as "lastStateUpdateDate" or something.

I think that makes sense. Should I create an issue for 8.0 release? This way we can make it a breaking change.

I think copying how it's done on create is ok for now but definitely strange.

Reply:
DateDateDate

I'm not against changing the name to better reflect reality, especially as an 8.0 thing.  Though I wonder if `lastStateUpdate` makes sense - the date value is supposed to reflect the last execution, except we knew we didn't have that date in some previous 7.x release, and obviously don't have one for newly created rules.  So it does kind of make sense to continue with the current name, with the caveat that you need to check to see if the `status` is 'pending' which means "it hasn't run yet".  Basically, if we call it `lastStateUpdate`, would folks expect other changes might also change this date?  What happens to these fields when disabling a rule?

Reply:
I read this as "the execution status is pending, and it's been pending since `lastExecutionDate`". So any name that can reflect that and last time the rule executed I'm +1. Seems like a separate issue to be created?